### PR TITLE
[PDK-421] Validate EPP syntax (WIP)

### PR DESCRIPTION
**Pull request by raphink**
This is a base implementation for validating EPP templates. I essentially copy/pasted the Puppet validate one, so it probably needs some pruning, but it works.

### CODE REVIEW

**Comment by scotje** [comment]
Hi @raphink! Thanks for the contribution! We'll get someone to take a look at this as soon as we can, in the meantime the test failures look like they should be pretty easy to address and the coveralls failure is just asking for this new code to have some unit test coverage.

We can look at addressing that ourselves but if you wanted to work on fixing up those failures it would help us get this reviewed and merged more quickly. :)

Thanks again!

File: spec/acceptance/validate_puppet_spec.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ class foo {
         its(:exit_status) { is_expected.to eq(0) }
         its(:stderr) { is_expected.to match(syntax_spinner_text) }
         its(:stderr) { is_expected.to match(lint_spinner_text) }
-        its(:stdout) { is_expected.to have_no_output }
+        its(:stdout) { is_expected.to match(%r{Target does not contain any files to validate}) }
```
Comment (glennsarti) [inline spec/acceptance/validate_puppet_spec.rb]: It seems very strange that adding a completely new class, which doesn't affect `.pp` files, and no test fixture changes, has caused this test to be modified

Comment (raphink) [reply spec/acceptance/validate_puppet_spec.rb]: The full message says it does not contain any EPP files to validate.

**Comment by raphink** [comment]
Just added some unit tests. I was wondering about the Puppet version contexts, but it does actually impact the EPP validation output as well.

**Comment by raphink** [comment]
The tests are passing now. Do you want me to add something still @glennsarti ?

**Comment by glennsarti** [comment]
@raphink Sorry for the delayed response. We're just in the middle of a PDK release, but I promise I'll get back to you soon.

File: spec/unit/pdk/validate/puppet/puppet_epp_spec.rb [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    expect(described_class).to have_attributes(
+      name:         'puppet-epp',
+      cmd:          'puppet',
+      spinner_text: a_string_matching(%r{puppet EPP syntax}i),
+    )
+  end
+
+  it_behaves_like 'it accepts .epp targets'
+
+  describe '.pattern_ignore' do
```
Comment (glennsarti) [inline spec/unit/pdk/validate/puppet/puppet_epp_spec.rb]: This test can be removed (I assume it's copy-paste from the .pp tests).  Puppet plans can never end in .epp

File: spec/unit/pdk/validate/puppet/puppet_epp_spec.rb [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+    it 'appends the targets to the command arguments' do
+      expect(command_args.last(targets.count)).to eq(targets)
+    end
+
+    context 'when auto-correct is enabled' do
+      let(:options) { { auto_correct: true } }
+
+      it 'has no effect' do
+        expect(command_args).to eq(%w[epp validate --config /dev/null --modulepath].push(tmpdir).concat(targets))
```
Comment (glennsarti) [inline spec/unit/pdk/validate/puppet/puppet_epp_spec.rb]: Note for me, not to stop merging:  Need to investigate why using a fixed Linux path.  This shouldn't work on Windows?

File: spec/acceptance/validate_puppet_spec.rb (hunk 1 above)
Comment (glennsarti) [reply spec/acceptance/validate_puppet_spec.rb]: Fair enough.

**Comment by glennsarti** [comment]
@rodjek Apart from removing one test I'm happy to merge this.  Do you want to review prior?

**Comment by raphink** [comment]
Done removing the extra test @scotje

**Comment by glennsarti** [comment]
All concerns have been addressed.  Merging.  Thanks again @raphink !!

**Pull request merged**